testsuite/tests/singlecluster/identical_hostnames/auth/test_identical_hostnames_auth_on_routes.py

```diff
@@ -0,0 +1,65 @@
+"""
+There used to be a limitation if using two HTTPRoutes declaring the same hostname related to AuthPolicy:
+https://github.com/Kuadrant/kuadrant-operator/blob/c8d083808daff46772254e223407c849b55020a7/doc/auth.md#limitation-multiple-network-resources-with-identical-hostnames
+(the first topology mentioned there)
+This test validates that it has been properly fixed, i.e. both AuthPolicies are fully and successfully enforced.
+"""
+
+import pytest
+
+from testsuite.kuadrant.policy.authorization.auth_policy import AuthPolicy
+
+pytestmark = [pytest.mark.kuadrant_only]
+
+
+@pytest.fixture(scope="module")
+def authorization2(route2, blame, cluster, label):
```

Comment:
This should also be in auth conftest

Reply:
Same as above. For one test this auth policy is attached to HTTPRoute, for the other it is attached to GW. Again, not sure how to handle this nicely so that it can be shared in both tests.